### PR TITLE
feat(GKE): add reservation samples

### PR DESCRIPTION
**Pull request by apeabody**
## Description

add GKE reservation samples

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/kubernetes-engine/docs/how-to/consuming-reservations

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [x] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [x] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

File: gke/autopilot/reservation/main.tf [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  }
+
+  specific_reservation_required = true
+}
+# [END gke_autopilot_reservation_specific_reservation]
+
+# [START gke_autopilot_reservation_specific_cluster]
+resource "google_compute_reservation" "specific_accelerator" {
+  name = "specific-reservation-accelerator"
+  zone = "us-central1-a"
```
Comment (yliaog) [inline gke/autopilot/reservation/main.tf:55]: the cluster is regional, but the reservation is for a zone, not sure if it works.

File: gke/standard/zonal/reservation/main.tf [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  # Set `deletion_protection` to `true` will ensure that one cannot
+  # accidentally delete this instance by use of Terraform.
+  deletion_protection = false
+}
+# [END gke_standard_zonal_reservation_any_cluster]
+
+# [START gke_standard_zonal_reservation_any_node_pool]
+resource "google_container_node_pool" "any-node_pool" {
+  name     = "gke-standard-regional-any-node-pool"
```
Comment (yliaog) [inline gke/standard/zonal/reservation/main.tf:59]: typo? line 57: says it is zonal, line 59 says it is regional

File: gke/standard/zonal/reservation/main.tf [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  }
+
+  specific_reservation_required = true
+}
+# [END gke_standard_zonal_reservation_specific_reservation]
+
+# [START gke_standard_zonal_reservation_specific_node_pool]
+resource "google_container_node_pool" "specific-node-pool" {
+  name     = "gke-standard-regional-specific-node-pool"
```
Comment (yliaog) [inline gke/standard/zonal/reservation/main.tf:93]: ditto

File: gke/standard/zonal/reservation/main.tf (hunk 2 above)
Comment (apeabody) [reply gke/standard/zonal/reservation/main.tf:59]: Yup - Thought I caught them all, but apparently not. :)

File: gke/standard/zonal/reservation/main.tf (hunk 3 above)
Comment (apeabody) [reply gke/standard/zonal/reservation/main.tf]: fixed

File: gke/standard/zonal/reservation/main.tf (hunk 2 above)
Comment (apeabody) [reply gke/standard/zonal/reservation/main.tf]: fixed

File: gke/autopilot/reservation/main.tf (hunk 1 above)
Comment (apeabody) [reply gke/autopilot/reservation/main.tf:55]: Good Catch @yliaog! - Worked fine during testing, but I'll adjust for consistency and re-test.

Comment (apeabody) [reply gke/autopilot/reservation/main.tf:55]: OK, so `google_compute_reservation` is always zone specific, and Autopilot clusters are regional, so this appears to be working correctly/intended.

**Review by yliaog — APPROVED** [review]

**Review by glasnt — APPROVED** [review]

**Comment by glasnt** [comment]
Unrelated CI failure. Will try re-running.

**Comment by apeabody** [comment]
(re-triggered from another flake)

**Comment by apeabody** [comment]
The repeated CI failure is from an unrelated GKE sample.  Moved my experiment on finer test filtering to https://github.com/terraform-google-modules/terraform-docs-samples/pull/669

**Pull request merged**